test/box-luatest/gh_7939_fix_meamleak_on_tuple_encode_test.lua

```diff
@@ -0,0 +1,187 @@
+                              'SELECT * from TEST', trouble)
+    c:ping()
+end
+
+-- Currently we don't have Lua API to check slab cache usage and we
+-- can thus we cannot check it for leaks. At least check error paths work.
+--
+-- TODO(gh-8677) If in the scope of the above issue an assertion will be added
+-- to the `cord_buf_on_yield` then we can test for leaks just by calling
+-- fiber.yield.
+g.test_misc_cord_ibuf = function(cg)
+    cg.server:exec(function()
+        local bad = require('ffi').cast('void *', -1)
+        local flush = string.rep('x', 1024 * 1024)
+        local trouble = {1, flush, bad}
+        local errmsg = "unsupported Lua type 'cdata'"
+        local msgpack = require('msgpack')
+
+        -- This test does not work with cord ibuf. Let it be here though.
+        local buffer = require('buffer')
+        local buf = buffer.ibuf()
+        msgpack.encode('something', buf)
+        local before = buf:size()
+        t.assert_error_msg_equals(errmsg, msgpack.encode, trouble, buf)
+        t.assert_equals(buf:size(), before)
+
+        t.assert_error_msg_equals(errmsg, msgpack.encode, trouble)
+        t.assert_error_msg_equals(errmsg, msgpack.object, trouble)
+        t.assert_error_msg_equals(errmsg, box.tuple.new, trouble)
+        local tuple = box.tuple.new{1, 2, 3}
+        t.assert_error_msg_equals(errmsg, box.tuple.transform,
+                                  tuple, 2, 1, trouble)
+        t.assert_error_msg_equals(errmsg, box.broadcast, 'key', trouble)
+    end)
+end
+
+g.test_misc_region_leaks = function(cg)
+    cg.server:exec(function()
+        local fiber = require('fiber')
+        local bad = require('ffi').cast('void *', -1)
+        local flush = string.rep('x', 1024 * 1024)
+        local trouble = {1, flush, bad}
+        local errmsg = "unsupported Lua type 'cdata'"
+        local mem_used = function()
+            return fiber.info()[fiber.self().id()].memory.used
+        end
+
+        local before = mem_used()
+        t.assert_error_msg_equals(errmsg, box.iproto.send, box.session.id(),
+                                  trouble, {})
+        t.assert_equals(before, mem_used())
+        t.assert_error_msg_equals(errmsg, box.iproto.send, box.session.id(),
+                                  {}, trouble)
+        t.assert_equals(before, mem_used())
+
+        local ret, err = box.execute([[SELECT lua('
+            return {
+                1,
+                string.rep("x", 1024 * 1024),
+                require("ffi").cast("void *", -1)
+            }
+        ');]])
+        t.assert_equals(ret, nil)
+        t.assert_equals(tostring(err), errmsg)
+        t.assert_equals(before, mem_used())
+
+        ret, err = box.execute('SELECT ?;', {{trouble}})
+        t.assert_equals(ret, nil)
+        t.assert_equals(tostring(err), errmsg)
+        t.assert_equals(before, mem_used())
+    end)
+end
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
```

Comment:
Nit: Please move the `after_all` hander to the top, near `before_all` and `after_each`.

Reply:
Done.